src/app/clusters/ota-requestor/OTARequestor.h

```diff
@@ -152,21 +159,39 @@ class OTARequestor : public OTARequestorInterface
         chip::BDXDownloader * mDownloader;
     };
 
-    // Variables
-    // TODO: align on variable naming standard
-    OTARequestorDriver * mOtaRequestorDriver;
-    chip::NodeId mProviderNodeId;
-    chip::FabricIndex mProviderFabricIndex;
-    uint32_t mOtaStartDelayMs                      = 0;
-    chip::CASESessionManager * mCASESessionManager = nullptr;
-    OnConnectedState onConnectedState              = kQueryImage;
-    chip::Messaging::ExchangeContext * exchangeCtx = nullptr;
-    chip::BDXDownloader * mBdxDownloader; // TODO: this should be OTADownloader
-    BDXMessenger mBdxMessenger;           // TODO: ideally this is held by the application
-
-    // TODO: Temporary until IP address resolution is implemented in the Exchange layer
-    chip::Inet::IPAddress mIpAddress;
-
-    // Functions
+    /**
+     * Setup CASESessionManager used to establish a session with the provider
+     */
     CHIP_ERROR SetupCASESessionManager(chip::FabricIndex fabricIndex);
+
+    /**
+     * Validate the URI and parse the BDX URI for various fields
+     */
+    CHIP_ERROR ParseBdxUri(CharSpan uri, NodeId & nodeId, CharSpan & fileDesignator);
```

Comment:
This could be a logic-only function in a separate file, as it is not dependent on any class state

Reply:
Actually, when I first started working on this, Trevor's change for BDXDownloader didn't merge it. Now that it's there, I can move this over to that class. Makes more sense there then here.

Reply:
As long as it can be unit tested soon, it can be put anywhere.

Reply:
Moved to BDXDownloader